pkg/alluxiocluster/delete_alluxiocluster.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxiocluster
+
+import (
+	"os"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	alluxiov1alpha1 "github.com/alluxio/k8s-operator/api/v1alpha1"
+	"github.com/alluxio/k8s-operator/pkg/logger"
+	"github.com/alluxio/k8s-operator/pkg/utils"
+)
+
+func DeleteAlluxioClusterIfExist(ctx AlluxioClusterReconcileReqCtx) (ctrl.Result, error) {
+	logger.Infof("Uninstalling Alluxio cluster %s.", ctx.NamespacedName.String())
+
+	helmCtx := utils.HelmContext{
+		ReleaseName: ctx.Name,
+		Namespace:   ctx.Namespace,
+	}
+	if err := utils.HelmDeleteIfExist(helmCtx); err != nil {
+		logger.Errorf("failed to delete helm release %s: %v", ctx.NamespacedName.String(), err)
+		return ctrl.Result{}, err
+	}
+
+	ctx.Dataset.Status.Phase = alluxiov1alpha1.DatasetPhasePending
```

Comment:
i think it's not easy to infer the status handling from the function name. maybe we can change the function? or we can move the status handling outside, although it might be a little verbose

Reply:
Makes sense.